.github/workflows/run_tests.yml

```diff
@@ -116,3 +117,40 @@ jobs:
         uses: codecov/codecov-action@v2
         with:
           files: ./tests/cpp/unit-tests/build/CMakeFiles/cpp_unit_tests.dir/coverage.info
+
+      - name: Install docker and redis-server
+        run: |
+          sudo apt-get update && sudo apt-get -y install curl gnupg lsb-release software-properties-common ca-certificates && \
+          # Add latest redis to apt sources
+          echo "deb [signed-by=/usr/share/keyrings/redis-archive-keyring.gpg] https://packages.redis.io/deb $(lsb_release -cs) main" | sudo tee /etc/apt/sources.list.d/redis.list && \
+          curl -fsSL https://packages.redis.io/gpg | gpg --dearmor -o /usr/share/keyrings/redis-archive-keyring.gpg && \
+          # Add latest docker to apt sources
+          echo  "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null && \
+          curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /usr/share/keyrings/docker-archive-keyring.gpg && \
+          sudo apt-get update && \
+          sudo apt-get -y install iputils-ping docker-ce docker-ce-cli containerd.io redis-server
```

Comment:
This might be out of the scope of this PR, but just wanted to kick off a discussion of whether we should be using a specific version of redis in our tests so we can more intentionally change version numbers to match SmartSim.  I see advantages to both options of pinning to a version or just using latest.  I think SmartSim uses 6.0.8 and I think this is pulling down 6.2.6.  This doesn't need to hold up this PR, we can create a follow on issue if we decide to match the smartsim version.

Reply:
After thinking more about this, it might make more sense to make the redis-server version identical to the one used in the redisai docker image so that the non-cluster and cluster images are more consistent.